File: src/main/java/nz/laspruca/tcplugin/eventlistners/PlayerJoinLeave.java
package nz.laspruca.tcplugin.eventlistners;

import nz.laspruca.tcplugin.commands.DiscordCommand;
import org.bukkit.event.EventHandler;
import org.bukkit.event.Listener;
import org.bukkit.event.player.PlayerJoinEvent;
import org.bukkit.event.player.PlayerQuitEvent;
import nz.laspruca.tcplugin.logger.loggerevent.*;

import static nz.laspruca.tcplugin.Plugin.*;

public class PlayerJoinLeave implements Listener {

	@EventHandler
	public void onPlayerJoin(PlayerJoinEvent event) {
		new JoinEvent(event.getPlayer().getDisplayName()).log();
		if (discord.loggedIn()) {
			if (!discord.getMembers().contains(event.getPlayer().getDisplayName())) {
				event.getPlayer().spigot().sendMessage(DiscordCommand.generateText());
			}
		}
	}

	@EventHandler
	public void onLeaveJoin(PlayerQuitEvent event) {
		new LeaveEvent(event.getPlayer().getDisplayName()).log();
	}
}

File: src/main/kotlin/nz/laspruca/tcplugin/eventListeners/PlayerJoinLeave.kt
package nz.laspruca.tcplugin.eventListeners

import nz.laspruca.tcplugin.logger.LeaveEvent
import org.bukkit.event.player.PlayerQuitEvent
import nz.laspruca.tcplugin.logger.JoinEvent
import org.bukkit.event.EventHandler
import org.bukkit.event.Listener
import org.bukkit.event.player.PlayerJoinEvent


object PlayerJoinLeave : Listener {
    @EventHandler
    fun onPlayerJoin(event: PlayerJoinEvent) {
        JoinEvent(event.player.name).log()
//        if (discord?.loggedIn!!) {
//            if (!discord!!.getMembers().contains(event.player.name)) {
//                event.player.sendMessage(generateText())
//            }
//        }
    }

    @EventHandler
    fun onLeaveJoin(event: PlayerQuitEvent) {
        LeaveEvent(event.player.name).log()
    }
}
